### PR TITLE
fix getDBWithName db.name pass from GetDatabaseByName

### PR DESCRIPTION
**Pull request by pinikomarov**

### CODE REVIEW

**Comment by pinikomarov** [comment]
This is a dependency for : add_service_db_finalizer_logic - https://github.com/openstack-k8s-operators/nova-operator/pull/174

File: modules/database/funcs.go [hunk 1]
```diff
@@ -316,7 +316,7 @@ func GetDatabaseByName(
 ) (*Database, error) {
 	// create a Database by suppplying a resource name
 	db := &Database{
-		databaseName: name,
+		name: name,
```
Comment (gibizer) [inline modules/database/funcs.go:319]: I quickly checked at it seems this will be compatible at least with the existing keystone code. But I suggest to check the rest of the operators calling `GetDatabaseByName` to see if they consistently using instance.Name as the MariaDBDatabase.Name.

I will go and do a nova build and deploy with this change and with the related nova PR and will report back.

Comment (gibizer) [reply modules/database/funcs.go:319]: Nova seems to work with this change as expected. So from nova-perspective it is good to go. 
If you checked the rest of the operators as per

> But I suggest to check the rest of the operators calling GetDatabaseByName to see if they consistently using instance.Name as the MariaDBDatabase.Name.

then let me know and I can approve this.

Comment (pinikomarov) [reply modules/database/funcs.go:319]: let me run that check first with the rest of them , I'll make it a draft for now

Comment (pinikomarov) [reply modules/database/funcs.go:319]: @gibizer I'm considering how best to test this and how to batch test this across all operators if I can.

1) to test this for instance on the cinder operator , I can run cinder operator on a running crc cluster via "make run" the add the CR and delete it and , while adding a logging line in the code checking what value -is used in instance.Name as the MariaDBDatabase.Name ? sounds about right ? how are you testing this ? 

2) any batch way to run install_yamls with all operators with a custom lib-common lib version or location and then start deleting existing CRs? 

What' could be an efficient approach here ?

Comment (gibizer) [reply modules/database/funcs.go:319]: 2 is the ultimate test but we don't have automation yet that helps bumping the lib-common version in each service opeartor, building that service operator and building an openstack-operator that embeds the fresh service operator versions. So this is a lot of manual work today.

So 1 could be an option. Especially that you don't have to really build anything. Just deploy with install_yamls and check what are the MariaDBDatabase.Name for each service there and also check the code in the service operator how `GetDatabaseByName` is called there to see that it is called would be called with the same value.

Comment (gibizer) [reply modules/database/funcs.go:319]:
```console
$ oc get MariaDBDatabase -o custom-columns=crName:.metadata.name,dbName:.spec.name,ownerKind:.metadata.ownerReferences[0].kind,ownerCRName:.metadata.ownerReferences[0].name
crName       dbName       ownerKind      ownerCRName
cinder       cinder       Cinder         cinder
glance       glance       Glance         glance
keystone     keystone     KeystoneAPI    keystone
neutron      neutron      NeutronAPI     neutron
nova-api     nova_api     Nova           nova
nova-cell0   nova_cell0   Nova           nova
placement    placement    PlacementAPI   placement
```
At least the name of the MariaDBDatabase CR the dbName in its Spec and the name of the owner CR seems to be equal (outside of nova) in the default deployment. But the code of the service operators still needs to be checked that they use the same logical thing (i.e. the name of the CR?) as the name of MariaDBDatabase CR when creating that and not something else, i.e. a hardcoded string.

Comment (pinikomarov) [reply modules/database/funcs.go:319]: Hi so I used this short script to trace back the in use CR tamplates from install_yamls's Makefile to the source repos using this short script: 
http://pastebin.test.redhat.com/1089554
and checked the naming of the CRs , not sure about the naming though.
what I should I verify on the crs, that the name: field corresonds with the dbuser ? or the name of the operator ? 

Does  that checkout with you  ? 
attaching fields below ..
```

./openstack-operator/config/samples/core_v1beta1_openstackcontrolplane.yaml
apiVersion: core.openstack.org/v1beta1
kind: OpenStackControlPlane
metadata:
  name: openstack
spec:
  secret: osp-secret
  storageClass: local-storage
      databaseInstance: openstack

EndFile

./keystone-operator/config/samples/keystone_v1beta1_keystoneapi.yaml
apiVersion: keystone.openstack.org/v1beta1
kind: KeystoneAPI
metadata:
  name: keystone
spec:
  adminProject: admin
  adminRole: admin
  databaseInstance: openstack

EndFile

./mariadb-operator/config/samples/mariadb_v1beta1_mariadb.yaml
apiVersion: mariadb.openstack.org/v1beta1
kind: MariaDB
metadata:
  name: openstack
spec:
  secret: mariadb-secret
  storageClass: local-storage

EndFile

./galera-operator/config/samples/mariadb_v1beta1_mariadb.yaml
apiVersion: mariadb.openstack.org/v1beta1
kind: MariaDB
metadata:
  name: openstack
spec:
  secret: mariadb-secret
  storageClass: local-storage

EndFile

./placement-operator/config/samples/placement_v1beta1_placementapi.yaml
apiVersion: placement.openstack.org/v1beta1
kind: PlacementAPI
metadata:
  name: placement
spec:
  serviceUser: placement
  containerImage: quay.io/tripleowallabycentos9/openstack-placement-api:current-tripleo
  databaseInstance: openstack

EndFile

./glance-operator/config/samples/glance_v1beta1_glance.yaml
apiVersion: glance.openstack.org/v1beta1
kind: Glance
metadata:
  name: glance
spec:
  serviceUser: glance
  containerImage: quay.io/tripleozedcentos9/openstack-glance-api:current-tripleo
  databaseInstance: openstack

EndFile

./ovn-operator/config/samples/ovn_v1beta1_ovndbcluster.yaml
apiVersion: ovn.openstack.org/v1beta1
kind: OVNDBCluster
metadata:
  name: ovndbcluster-nb-sample
spec:
  replicas: 1
  containerImage: quay.io/tripleozedcentos9/openstack-ovn-nb-db-server:current-tripleo
--
apiVersion: ovn.openstack.org/v1beta1
kind: OVNDBCluster
metadata:
  name: ovndbcluster-sb-sample
spec:
  replicas: 1
  containerImage: quay.io/tripleozedcentos9/openstack-ovn-sb-db-server:current-tripleo

EndFile

./ovn-operator/config/samples/ovn_v1beta1_ovnnorthd.yaml
apiVersion: ovn.openstack.org/v1beta1
kind: OVNNorthd
metadata:
  name: ovnnorthd-sample
spec:
  replicas: 1
  containerImage: quay.io/tripleozedcentos9/openstack-ovn-northd:current-tripleo

EndFile

./neutron-operator/config/samples/neutron_v1beta1_neutronapi.yaml
apiVersion: neutron.openstack.org/v1beta1
kind: NeutronAPI
metadata:
  name: neutron
  namespace: openstack
spec:
  serviceUser: neutron
  databaseInstance: openstack

EndFile

./cinder-operator/config/samples/cinder_v1beta1_cinder.yaml
apiVersion: cinder.openstack.org/v1beta1
kind: Cinder
metadata:
  name: cinder
  namespace: openstack
spec:
  serviceUser: cinder
  databaseInstance: openstack

EndFile

./rabbitmq-cluster-operator/docs/examples/default-security-context/rabbitmq.yaml
apiVersion: rabbitmq.com/v1beta1
kind: RabbitmqCluster
metadata:
  name: default-security-context
spec:
  override:
    statefulSet:
--
            securityContext: {}
            containers: []
            initContainers:
            - name: setup-container
              securityContext: {}

EndFile

./ironic-operator/config/samples/ironic_v1beta1_ironic.yaml
apiVersion: ironic.openstack.org/v1beta1
kind: Ironic
metadata:
  name: ironic
  namespace: openstack
spec:
  serviceUser: ironic
  databaseInstance: openstack

EndFile

./octavia-operator/config/samples/octavia_v1beta1_octaviaapi.yaml
apiVersion: octavia.openstack.org/v1beta1
kind: OctaviaAPI
metadata:
  name: octavia
spec:
  # TODO(tweining): Add fields here
  databaseInstance: openstack
  databaseInstance: openstack

EndFile

./nova-operator/config/samples/nova_v1beta1_nova.yaml
apiVersion: nova.openstack.org/v1beta1
kind: Nova
metadata:
  name: nova
spec:
  secret: osp-secret
  cellTemplates:

EndFile

```

Comment (gibizer) [reply modules/database/funcs.go:319]: > Hi so I used this short script to trace back the in use CR tamplates from install_yamls's Makefile to the source repos using this short script: http://pastebin.test.redhat.com/1089554 and checked the naming of the CRs , not sure about the naming though. what I should I verify on the crs, that the name: field corresonds with the dbuser ? or the name of the operator ?

I think the CR naming in the samples are not important if the below investigation checks out.

> 
> the GetDatabaseByName functions all look ok to me:
> 
> ```
> [r@r openstack-k8s-operators]$ grep -r GetDatabaseByName *
> cinder-operator/controllers/cinder_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> designate-operator/controllers/designateapi_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> glance-operator/controllers/glance_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> heat-operator/controllers/heat_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> ironic-operator/controllers/ironic_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> keystone-operator/controllers/keystoneapi_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> neutron-operator/controllers/neutronapi_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> octavia-operator/controllers/octaviaapi_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> placement-operator/controllers/placementapi_controller.go:	db, err := database.GetDatabaseByName(ctx, helper, instance.Name)
> ```
> 

OK so this show that all the operator uses the `instance.Name` when query the MariaDBDatabase. Now we need to check that this matches with the information that is used when the operator creates the MariaDBDatabase via lib common's NewDatabase* functions. Note that `NewDatabase` does not allow providing the name for the MAriaDBDatabase. Instead lib-common will use the `instance.Name` from the instance stored in the Helper. So that is the simple case. But NewDatabaseWithNamespace allows passing in the name explicitly so there we need to check if the same `instance.Name` is passed in as the used at `GetDatabaseByName`. 

I went and checked the operators:
* cinder: https://github.com/openstack-k8s-operators/cinder-operator/blob/ad6adb43ab7f279c24112632d9a39a8b7374c783/controllers/cinder_controller.go#L248-L260 -> NewDatabase is used so lib-common uses `instance.Name` automatically. So it is OK.
* designate https://github.com/openstack-k8s-operators/designate-operator/blob/b3ade480899650d3ef015d4531d6663d0bababd5/controllers/designateapi_controller.go#L253-L265 uses NewDatabase -> OK
* glance: https://github.com/openstack-k8s-operators/glance-operator/blob/a7c3745ebcf59c1f48dc6f1438384f646aefd42a/controllers/glance_controller.go#L282-L294 uses NewDatabase -> OK
* https://github.com/openstack-k8s-operators/heat-operator/blob/a3b3b3677e5e9f53ca2bb89aa65fbf694c1492d4/controllers/heat_controller.go#L374-L386 uses NewDatabase -> OK
* ironic: https://github.com/openstack-k8s-operators/ironic-operator/blob/fdf999999dfd56ddcf2d41143b0a077708db791a/controllers/ironic_controller.go#L426-L438 uses NewDatabase -> OK
* keystone: https://github.com/openstack-k8s-operators/keystone-operator/blob/fdba0b3df1ca63f14d62b2fe66be5d22f3519c42/controllers/keystoneapi_controller.go#L225-L237 uses NewDatabse -> OK
* neutron: https://github.com/openstack-k8s-operators/neutron-operator/blob/7a31f3b0f8f22abeafc31329d4af094363fe9f2d/controllers/neutronapi_controller.go#L266-L278 uses NewDatabase -> OK
* octavia: https://github.com/openstack-k8s-operators/octavia-operator/blob/a433150c957f9be195efb1a069a34b2c0ec8e84b/controllers/octaviaapi_controller.go#L254-L266 uses NewDatabase -> OK
* placement: https://github.com/openstack-k8s-operators/placement-operator/blob/b33eabf1a8bfce3607452691cfc6ecd9926e543e/controllers/placementapi_controller.go#L265-L277 uses NewDatabase -> OK

So it seems only nova uses NewDatabaseWithNamespace:
* https://github.com/openstack-k8s-operators/nova-operator/blob/b0e3f90e8b15c4009c873181a847c6af7e48721a/controllers/nova_controller.go#L559
* https://github.com/openstack-k8s-operators/nova-operator/blob/b0e3f90e8b15c4009c873181a847c6af7e48721a/controllers/nova_controller.go#L587
But we checked those already. 

Bottom line I think we are good to go.

**Review by gibizer — APPROVED** [review]

**Comment by gibizer** [comment]
Thanks @pinikomarov !
